inc/Engine/Common/PerformanceHints/Frontend/Processor.php

```diff
@@ -59,17 +59,27 @@ public function maybe_apply_optimizations( string $html ): string {
 		$is_mobile = $this->is_mobile();
 
 		$html_optimized = null;
+		// Flag to check if any optimization has been applied.
+		$optimization_applied = false;
+
 		foreach ( $this->factories as $factory ) {
 			$row = $factory->queries()->get_row( $url, $is_mobile );
-			if ( empty( $row ) ) {
-				return $this->inject_beacon( $html, $url, $is_mobile );
+
+			if ( ! empty( $row ) ) {
+				$html_optimized = $factory->get_frontend_controller()->optimize( $html, $row );
+				// Update html for the next iteration.
+				$html = $html_optimized;
+				// Set flag as true since optimization has been applied.
+				$optimization_applied = true;
 			}
```

Comment:
```suggestion
			if ( empty( $row ) ) {
				continue;
			}
			$html_optimized = $factory->get_frontend_controller()->optimize( $html, $row );
			// Update html for the next iteration.
			$html = $html_optimized;
			// Set flag as true since optimization has been applied.
			$optimization_applied = true;
```